src/Nethermind/Nethermind.Merge.Plugin/Handlers/V1/NewPayloadV1Handler.cs

```diff
@@ -362,28 +361,26 @@ void GetProcessingQueueOnBlockRemoved(object? o, BlockHashEventArgs e)
 
                 if (!result.HasValue)
                 {
+                    // we don't know the result of processing the block, either because
+                    // it is the first time we add it to the tree or it's AlreadyKnown in
+                    // the tree but hasn't yet been processed. if it's the second case
+                    // probably the block is already in the processing queue as a result
+                    // of a previous newPayload or the block being discovered during syncing
+                    // but add it to the processing queue just in case.
                     _processingQueue.Enqueue(block, processingOptions);
-
                     result = await blockProcessed.TimeoutOn(timeoutTask);
                 }
-
-                if ((result & ValidationResult.Valid) == 0 && (result & ValidationResult.Syncing) == 0)
-                {
-                    if (_logger.IsWarn) _logger.Warn($"Block {block.ToString(Block.Format.FullHashAndNumber)} cannot be processed and wont be accepted to the tree.");
-                }
             }
             catch (TimeoutException)
             {
                 if (_logger.IsDebug) _logger.Debug($"Block {block.ToString(Block.Format.FullHashAndNumber)} timed out when processing. Assume Syncing.");
-                result = ValidationResult.Syncing;
```

Comment:
I think this is needed.

Reply:
Nevermind, result would be non-null.

Reply:
Maybe it's worth adding a comment in the code